### ISSUE TITLE
[DragTarget] The following _CastError was thrown while routing a pointer event: Null check operator used on a null value

### ISSUE THREAD

**Issue by ksiventy** [flutter/flutter#106097]
```
[+4219 ms] ══╡ EXCEPTION CAUGHT BY GESTURE LIBRARY ╞═══════════════════════════════════════════════════════════
[        ] The following TypeErrorImpl was thrown while routing a pointer event:
[        ] Unexpected null value.
[        ] When the exception was thrown, this was the stack:
[        ] dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 251:49      throw_
[        ] dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 528:63  nullCheck
[        ] packages/flutter/src/widgets/drag_target.dart 778:76                              didDrop
[        ] packages/flutter/src/widgets/drag_target.dart 939:7                               finishDrag
[        ] packages/flutter/src/widgets/drag_target.dart 855:5                               end
[        ] packages/flutter/src/gestures/multidrag.dart 151:13                               [_up]
[        ] packages/flutter/src/gestures/multidrag.dart 263:12                               [_handleEvent]
[        ] packages/flutter/src/gestures/pointer_router.dart 94:12                           [_dispatch]
[        ] packages/flutter/src/gestures/pointer_router.dart 139:9                           <fn>
[        ] dart-sdk/lib/_internal/js_dev_runtime/private/linked_hash_map.dart 21:13          forEach
[        ] packages/flutter/src/gestures/pointer_router.dart 137:17                          [_dispatchEventToRoutes]
[        ] packages/flutter/src/gestures/pointer_router.dart 123:7                           route
[        ] packages/flutter/src/gestures/binding.dart 445:19                                 handleEvent
[        ] packages/flutter/src/gestures/binding.dart 425:14                                 dispatchEvent
[        ] packages/flutter/src/rendering/binding.dart 329:11                                dispatchEvent
[        ] packages/flutter/src/gestures/binding.dart 380:7                                  [_handlePointerEventImmediately]
[        ] packages/flutter/src/gestures/binding.dart 344:5                                  handlePointerEvent
[        ] packages/flutter/src/gestures/binding.dart 302:7                                  [_flushPointerEventQueue]
[        ] packages/flutter/src/gestures/binding.dart 285:32                                 [_handlePointerDataPacket]
[        ] lib/_engine/engine/platform_dispatcher.dart 1105:13                               invoke1
[        ] lib/_engine/engine/platform_dispatcher.dart 185:5                                 invokeOnPointerDataPacket
[        ] lib/_engine/engine/pointer_binding.dart 130:39                                    [_onPointerData]
[        ] lib/_engine/engine/pointer_binding.dart 543:18                                    <fn>
[        ] lib/_engine/engine/pointer_binding.dart 496:21                                    <fn>
[        ] lib/_engine/engine/pointer_binding.dart 210:16                                    loggedHandler
[        ] router: Instance of 'PointerRouter'
[        ] route: Closure: (PointerEvent) => void from: function () { [native code] }
[        ] event: PointerUpEvent#8c559(position: Offset(352.0, 428.7))
[        ] 
```
════════════════════════════════════════════════════════════════════════════════════════════════════
```dart
DragTarget(
                  builder: (context, list, data) {
                    return Text('test');
                  },
onAcceptWithDetails: (DragTargetDetails<String> details) {
                    debugPrint('onAcceptWithDetails ${details.data}');
                  },
)
```
Running Chrome
The Exception happened when I drag a draggable item within target

[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale zh-Hans-CN)
    • Flutter version 3.0.2 at /Users/niu/workspace/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (8 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

**Comment by maheshmnj** [comment]
Hi @ksiventy, Please share a [minimal and complete reproducible code](https://stackoverflow.com/help/minimal-reproducible-example) along with the output of `flutter doctor -v` to better address the issue.

**Comment by ksiventy** [comment]
flutter doctor -v
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale zh-Hans-CN)
    • Flutter version 3.0.2 at /Users/niu/workspace/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (8 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 102.0.5005.115

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
niu@yb-niudeMacBook-Air myapp % flutter doctor -v
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale zh-Hans-CN)
    • Flutter version 3.0.2 at /Users/niu/workspace/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (8 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/niu/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[!] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ✗ CocoaPods installed but not working.
        You appear to have CocoaPods installed but it is not working.
        This can happen if the version of Ruby that CocoaPods was installed with is different from the one being used to invoke it.
        This can usually be fixed by re-installing CocoaPods.
      To re-install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 102.0.5005.115

[✓] HTTP Host Availability
    • All required HTTP hosts are available

**Comment by ksiventy** [comment]
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  final SystemUiOverlayStyle _style =
      const SystemUiOverlayStyle(statusBarColor: Colors.transparent);

  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    SystemChrome.setSystemUIOverlayStyle(_style);
    return MaterialApp(
      home: const MainStatefulWidget(),
      theme: ThemeData(primarySwatch: Colors.blue),
    );
  }
}

class MainStatefulWidget extends StatefulWidget {
  const MainStatefulWidget({Key? key}) : super(key: key);

  @override
  State<StatefulWidget> createState() {
    return _MainState();
  }
}

class _MainState extends State<StatefulWidget> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("main"),
      ),
      body: Center(
        child: Row(
          children: [
            Expanded(
                child: Container(
              color: Colors.black26,
              child: Center(
                child: DragTarget(
                  builder: (context, list, data) {
                    // for (var value in list) {
                    //   debugPrint("list  $value");
                    // }
                    // for (var value in data) {
                    //   debugPrint("data  $value");
                    // }
                    return Text('adsfaf');
                  },
                  // onAccept: (w){
                  //   debugPrint("onAccept  $w");
                  // },
                  // onWillAccept:(w){
                  //   debugPrint("onWillAccept  $w");
                  //   return true;
                  // },
                  onAcceptWithDetails: (DragTargetDetails<String> details) {
                    debugPrint('onAcceptWithDetails ${details.data}');
                  },
                ),
              ),
            )),
            Column(
              children: [
                Draggable(
                    feedback: ElevatedButton(
                      onPressed: () {},
                      child: const Text('drag'),
                    ),
                    child: ElevatedButton(
                      onPressed: () {},
                      child: const Text('drag'),
                    ))
              ],
            )
          ],
        ),
      ),
    );
  }
}

**Comment by maheshmnj** [comment]
Hi @ksiventy, Thanks for filing the issue. I am able to reproduce the error on both stable and the master channel.

<details>
<summary>Android logs</summary>

```
Could not load custom device from config index 0: Expected enabled to be a boolean.
Running "flutter pub get" in sample...
Launching lib/main.dart on sdk gphone arm64 in debug mode...
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:58244/17Hy0uALTm8=/ws

════════ Exception caught by gesture library ═══════════════════════════════════
The following _CastError was thrown while routing a pointer event:
Null check operator used on a null value

When the exception was thrown, this was the stack
#0      _DragTargetState.didDrop
#1      _DragAvatar.finishDrag
#2      _DragAvatar.end
#3      MultiDragPointerState._up
#4      MultiDragGestureRecognizer._handleEvent
#5      PointerRouter._dispatch
#6      PointerRouter._dispatchEventToRoutes.<anonymous closure>
#7      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:614:13)
#8      PointerRouter._dispatchEventToRoutes
#9      PointerRouter.route
#10     GestureBinding.handleEvent
#11     GestureBinding.dispatchEvent
#12     RendererBinding.dispatchEvent
#13     GestureBinding._handlePointerEventImmediately
#14     GestureBinding.handlePointerEvent
#15     GestureBinding._flushPointerEventQueue
#16     GestureBinding._handlePointerDataPacket
#17     _invoke1 (dart:ui/hooks.dart:167:13)
#18     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:341:7)
#19     _dispatchPointerDataPacket (dart:ui/hooks.dart:94:31)
════════════════════════════════════════════════════════════════════════════════
```  
</details> 

<details>
<summary>Logs (web)</summary>

```
mahesh@Maheshs-MacBook-Air-M1 sample % flutter run -d chrome
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             12.3s
This app is linked to the debug service: ws://127.0.0.1:58414/KVUVf9QXJ-E=/ws
Debug service listening on ws://127.0.0.1:58414/KVUVf9QXJ-E=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:58414/KVUVf9QXJ-E=
Flutter Web Bootstrap: Programmatic
The Flutter DevTools debugger and profiler on Chrome is available at:
http://127.0.0.1:9101?uri=http://127.0.0.1:58414/KVUVf9QXJ-E=
══╡ EXCEPTION CAUGHT BY GESTURE LIBRARY ╞═══════════════════════════════════════════════════════════
The following TypeErrorImpl was thrown while routing a pointer event:
Unexpected null value.

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 266:49      throw_
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 528:63  nullCheck
packages/flutter/src/widgets/drag_target.dart 786:76                              didDrop
packages/flutter/src/widgets/drag_target.dart 954:7                               finishDrag
packages/flutter/src/widgets/drag_target.dart 867:5                               end
packages/flutter/src/gestures/multidrag.dart 152:13                               [_up]
packages/flutter/src/gestures/multidrag.dart 260:12                               [_handleEvent]
packages/flutter/src/gestures/pointer_router.dart 95:12                           [_dispatch]
packages/flutter/src/gestures/pointer_router.dart 140:9                           <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/linked_hash_map.dart 21:13          forEach
packages/flutter/src/gestures/pointer_router.dart 138:17                          [_dispatchEventToRoutes]
packages/flutter/src/gestures/pointer_router.dart 124:7                           route
packages/flutter/src/gestures/binding.dart 450:19                                 handleEvent
packages/flutter/src/gestures/binding.dart 430:14                                 dispatchEvent
packages/flutter/src/rendering/binding.dart 337:11                                dispatchEvent
packages/flutter/src/gestures/binding.dart 385:7                                  [_handlePointerEventImmediately]
packages/flutter/src/gestures/binding.dart 347:5                                  handlePointerEvent
packages/flutter/src/gestures/binding.dart 304:7                                  [_flushPointerEventQueue]
packages/flutter/src/gestures/binding.dart 285:7                                  [_handlePointerDataPacket]
lib/_engine/engine/platform_dispatcher.dart 1118:13                               invoke1
lib/_engine/engine/platform_dispatcher.dart 184:5                                 invokeOnPointerDataPacket
lib/_engine/engine/pointer_binding.dart 147:39                                    [_onPointerData]
lib/_engine/engine/pointer_binding.dart 647:18                                    <fn>
lib/_engine/engine/pointer_binding.dart 589:21                                    <fn>
lib/_engine/engine/pointer_binding.dart 286:16                                    loggedHandler

router: Instance of 'PointerRouter'
route: Closure: (PointerEvent) => void from: function () { [native code] }
event: PointerUpEvent#c5148(position: Offset(530.6, 388.4))
════════════════════════════════════════════════════════════════════════════════════════════════════
the Dart compiler exited unexpectedly.
```
</details> 

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.2 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (5 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1237, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.1.0-0.0.pre.1237 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6062e40510 (64 minutes ago), 2022-06-12 23:53:08 -0400
    • Engine revision f80294b57e
    • Dart version 2.18.0 (build 2.18.0-180.0.dev)
    • DevTools version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by TahaTesser** [comment]
This crash is due to `Draggable .data` not being initialized, a duplicate of https://github.com/flutter/flutter/issues/106097.


### Workaround

Provide a non-null value to `Draggable .data`.



<details> 
<summary>Code Sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(useMaterial3: true),
      darkTheme: ThemeData(useMaterial3: true),
      home: const Example(),
    );
  }
}

class Example extends StatelessWidget {
  const Example({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('DragTarget Sample'),
      ),
      body: Center(
        child: Row(
          children: <Widget>[
            Expanded(
                child: Container(
              color: Colors.black26,
              child: Center(
                child: DragTarget<int>(
                  builder: (BuildContext context, List<dynamic> list,
                      List<dynamic> data) {
                    return const Text('Target');
                  },
                  onAccept: (int? data) => print('onAccept $data'),
                ),
              ),
            )),
            Column(
              children: <Widget>[
                Draggable<String>(
                  data: '',
                  feedback: ElevatedButton(
                    onPressed: () {},
                    child: const Text('Drag'),
                  ),
                  child: ElevatedButton(
                    onPressed: () {},
                    child: const Text('Drag'),
                  ),
                )
              ],
            )
          ],
        ),
      ),
    );
  }
}

``` 
	
</details>
